tests/components/azure_data_explorer/test_init.py

```diff
@@ -0,0 +1,288 @@
+
+
+async def test_put_event_on_queue_with_queueing_client(
+    hass,
+    entry_queued,
+    mock_QueuedIngest,
+) -> None:
+    """Test listening to events from Hass. and writing to ADX with managed client."""
+
+    hass.states.async_set("sensor.test_sensor", STATE_ON)
+
+    async_fire_time_changed(
+        hass, utcnow() + timedelta(seconds=entry_queued.options[CONF_SEND_INTERVAL])
+    )
+
+    await hass.async_block_till_done()
+    mock_QueuedIngest.assert_called_once()
+    assert type(mock_QueuedIngest.call_args.args[0]) is StreamDescriptor
+
+
+async def test_import(hass) -> None:
+    """Test the popping of the filter and further import of the config."""
+    config = {
+        DOMAIN: {
+            "filter": {
+                "include_domains": ["light"],
+                "include_entity_globs": ["sensor.included_*"],
+                "include_entities": ["binary_sensor.included"],
+                "exclude_domains": ["light"],
+                "exclude_entity_globs": ["sensor.excluded_*"],
+                "exclude_entities": ["binary_sensor.excluded"],
+            },
+        }
+    }
+
+    assert await async_setup_component(hass, DOMAIN, config)
+    await hass.async_block_till_done()
+
+    assert "filter" in hass.data[DOMAIN]
+
+
+async def test_unload_entry(
+    hass,
+    entry_managed,
+    mock_ManagedStreaming,
+) -> None:
+    """Test being able to unload an entry.
+
+    Queue should be empty, so adding events to the batch should not be called,
+    this verifies that the unload, calls async_stop, which calls async_send and
+    shuts down the hub.
+    """
+    assert entry_managed.state == ConfigEntryState.LOADED
+    assert await hass.config_entries.async_unload(entry_managed.entry_id)
+    mock_ManagedStreaming.assert_not_called()
+    assert entry_managed.state == ConfigEntryState.NOT_LOADED
+
+
+@pytest.mark.freeze_time("2024-01-01 00:00:00")
+async def test_late_event(
+    hass,
+    entry_with_one_event,
+    mock_ManagedStreaming,
+) -> None:
+    """Test the check on late events."""
+    with patch(
+        f"{AZURE_DATA_EXPLORER_PATH}.utcnow",
+        return_value=utcnow() + timedelta(hours=1),
+    ):
+        async_fire_time_changed(hass, datetime(2024, 1, 2, 00, 00, 00))
+        await hass.async_block_till_done()
+        mock_ManagedStreaming.add.assert_not_called()
+
+
+@pytest.mark.parametrize(
+    ("filter_schema", "tests"),
+    [
+        (
+            {
+                "include_domains": ["light"],
+                "include_entity_globs": ["sensor.included_*"],
+                "include_entities": ["binary_sensor.included"],
+            },
+            [
+                FilterTest("climate.excluded", 0),
+                FilterTest("light.included", 1),
+                FilterTest("sensor.excluded_test", 0),
+                FilterTest("sensor.included_test", 1),
+                FilterTest("binary_sensor.included", 1),
+                FilterTest("binary_sensor.excluded", 0),
```

Comment:
```suggestion
                FilterTest("climate.excluded", expect_called=False),
                FilterTest("light.included", expect_called=True),
                FilterTest("sensor.excluded_test", expect_called=False),
                FilterTest("sensor.included_test", expect_called=True),
                FilterTest("binary_sensor.included", expect_called=True),
                FilterTest("binary_sensor.excluded", expect_called=False),
```

Please adopt the others